iosHyperskillApp/iosHyperskillApp/Sources/Modules/ProjectSelectionList/Views/Grid/ProjectSelectionListGridSectionView.swift

```diff
@@ -0,0 +1,110 @@
+import shared
+import SwiftUI
+
+extension ProjectSelectionListGridSectionView {
+    struct Appearance {
+        let rootSpacing: CGFloat = 16
+        let interitemSpacing: CGFloat = 8
+    }
+}
+
+struct ProjectSelectionListGridSectionView: View {
+    private(set) var appearance = Appearance()
+
+    let sectionTitle: String?
+    let sectionSubtitle: String?
+
+    let selectedProjectID: Int64?
+
+    let projects: [ProjectSelectionListFeature.ProjectListItem]
+    let projectsColumnsCount: Int
+
+    let onProjectTap: (Int64) -> Void
+
+    var body: some View {
+        if projects.isEmpty {
+            EmptyView()
+        } else {
+            VStack(alignment: .leading, spacing: appearance.rootSpacing) {
+                buildSectionHeader()
+                buildSectionContent()
+            }
+        }
+    }
+
+    @ViewBuilder
+    private func buildSectionHeader() -> some View {
+        if sectionTitle?.isEmpty ?? true && sectionSubtitle?.isEmpty ?? true {
+            EmptyView()
+        } else {
+            VStack(alignment: .leading, spacing: appearance.interitemSpacing) {
+                if let sectionTitle {
+                    Text(sectionTitle)
+                        .font(.title3)
+                        .bold()
+                        .foregroundColor(.primaryText)
+                }
+
+                if let sectionSubtitle {
+                    Text(sectionSubtitle)
+                        .font(.callout)
+                        .foregroundColor(.secondaryText)
+                }
+            }
+        }
+    }
+
+    @ViewBuilder
+    private func buildSectionContent() -> some View {
+        LazyVGrid(
+            columns: Array(
+                repeating: GridItem(
+                    .flexible(),
+                    spacing: appearance.interitemSpacing,
+                    alignment: .top
+                ),
+                count: projectsColumnsCount
+            ),
+            alignment: .leading,
+            spacing: appearance.interitemSpacing
+        ) {
+            ForEach(projects, id: \.id) { project in
+                ProjectSelectionListGridCellView(
+                    project: project,
+                    isSelected: project.id == selectedProjectID,
```

Comment:
Depending on design I suppose that selected project should be rendered first with no other projects and nowhere else. Here we can already detect that this section is responsible for rendering of selected project and pass `isSelectedProjectSection` param instead of `selectedProjectID`

Reply:
@vladkash in the first iteration - Yes, but in the future this could be changed. If you will look at the design, you can find project widgets with a `Selected` badge in other sections.

OK, for now, I'll turn off the `isSelected` behavior in the other sections.